services/blockstorage/adapter/test/contract_test.go

```diff
@@ -299,18 +299,17 @@ func newInMemoryAdapter(logger log.Logger) adapter.BlockPersistence {
 }
 
 func newFilesystemAdapter(logger log.Logger) (adapter.BlockPersistence, func()) {
-	ctx, cancel := context.WithCancel(context.Background())
-
 	conf := newTempFileConfig()
-	cleanup := func() {
-		cancel()
-		_ = os.RemoveAll(conf.BlockStorageFileSystemDataDir()) // ignore errors - nothing to do
-	}
 
-	persistence, err := filesystem.NewBlockPersistence(ctx, conf, logger, metric.NewRegistry())
+	persistence, err := filesystem.NewBlockPersistence(conf, logger, metric.NewRegistry())
 	if err != nil {
 		panic(err.Error())
 	}
 
+	cleanup := func() {
+		persistence.GracefulShutdown(context.Background())
```

Comment:
why not `ctx`? and if its not clear, why not `Todo()`?

Reply:
because there's no context here and I don't see the point in adding one just for a graceful shutdown call that doesn't actually wait on the context :)